flask_stupe/pagination.py

```diff
@@ -1,17 +1,42 @@
 import functools
 
 from flask import request
-
 from flask_stupe import pymongo
 
 __all__ = []
 
 
 if pymongo:
-    def _paginate(cursor, skip=None, limit=None, sort=None, count=True):
-        metadata = getattr(request, "metadata", None)
-        if count and isinstance(metadata, dict):
-            metadata.update(count=cursor.count())
+
+    def _get_link_header(base_url, total_count, limit, skip):
+        links = []
+        template = '<{}?limit={{limit}}&skip={{skip}}>; ' \
+                   'rel="{{rel}}"'.format(base_url)
+        prev_skip = skip - limit
+        next_skip = skip + limit
+        if (prev_skip >= 0):
+            k = "prev" if (prev_skip > limit) else "first"
+            links.append(template.format(limit=limit, skip=skip, rel=k))
+        if (next_skip < total_count):
+            k = "next" if (next_skip < (total_count - limit)) else "last"
+            links.append(template.format(limit=limit, skip=skip, rel=k))
+        return ", ".join(links)
+
+    def _set_pagination_headers(total_count=0, limit=0, skip=0):
+        headers = getattr(request, "response_headers")
+        headers["X-Total-Count"] = total_count
+        headers["Link"] = _get_link_header(request.base_url, total_count,
+                                           limit, skip)
+
+    def _paginate(cursor, skip=None, limit=None, sort=None, count=True,
+                  headers=False):
+
+        if headers:
+            _set_pagination_headers(cursor.count(), limit, skip)
```

Comment:
I feel it's not worth making a function to externalize those three lines, could you embed them here?